### ISSUE TITLE
Long load times on Sidewalk Gallery

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#2661]
When I visited Sidewalk Gallery on the Seattle server this morning, the landing page took over 30 seconds to load...

Has anyone else experienced this? I didn't noticed long load times on previous Sidewalk Gallery versions (previous to the Aug 7 release).

![image](https://user-images.githubusercontent.com/1621749/128759252-b07c150f-8f91-4816-a4a3-4b8da6b369ee.png)

**Comment by jonfroehlich** [comment]
Seattle is also taking quite a bit of time with some filter selections (e.g., switching the menu to show "Surface Problems")

**Comment by jonfroehlich** [comment]
SPGG is also taking a bit of time. Notably, I don't recall this delay on previous versions... but not sure why that would be affected by the most recent releases.

**Comment by misaugstad** [comment]
Hmm I might recall a change in the code that possibly increased the number of labels that we fetch from the server to help with paging..? I think it was something that i saw while reviewing code from @ThatOneGoat. @ThatOneGoat does that sound familiar?

**Comment by jonfroehlich** [comment]
Still an issue for Seattle, so probably a product of database size?

https://sidewalk-sea.cs.washington.edu/gallery

I also think that Seattle is one of the remaining servers not running Postgres 12 according to recent email thread with Jason Howe.

**Comment by misaugstad** [comment]
Agreed that those are the two factors here. We should be upgrading the Seattle database to postgres 12 tomorrow, and I'm interested to see if this is still an issue at that point!

**Comment by misaugstad** [comment]
I recently added a section to the user dashboard that shows a bunch of labeling mistakes in #2994. I had to do a similar thing on the back end as we do for Gallery: checking a bunch of pano IDs to see if GSV imagery is available in parallel, requiring a certain number for each label type.

The method I ended up using should be faster, though the code is a bit more complicated. But now that I've written it once, it should easily translate over to Gallery. So I want to take a quick stab at changing how we do it to see if it improves the load time at all.

**Comment by misaugstad** [comment]
I want to reopen this for now, because there is a pretty big improvement that I want to make here still. For any code that checks for GSV imagery, I'd like to skip the step where we check for imagery if we last checked very recently. Maybe that's in the past day or the past week or something. I'll create a new issue for this because it's even more likely to help out with loading the Validate page anyway.

In addition, we haven't actually tested that the page load times are good now :)

**Comment by misaugstad** [comment]
>there is a pretty big improvement that I want to make here still ... I'll create a new issue for this because it's even more likely to help out with loading the Validate page anyway

The new issue I made is #3004. Jon also commented on #3004 saying "FWIW, I've found that Gallery is actually quite performant". Given that the load times are actually somewhat reasonable now, and the only thing left to do here is contained in #3004, I'm just going to close this one in favor of that one!
